massgov/mayflower
# Feature/dp 9486  location filters

## Description
The location listing pattern Javascript has been updated to persist the form filters in the following scenarios (test steps below should cover all 4 scenarios):
1) A user comes to the location listing, fills in form filters, and submits the form
2) A user performs a search, clicks one of the result links which leaves the form page, and clicks the back button
3) A user performs a search with filters, and they de-select filters, the results should be updated instantly, and they should also be persisted if the user leaves the page and uses the back button to return to the page. The pagination should also be persisted.

## Related Issue / Ticket

- https://jira.mass.gov/browse/DP-9486 (Location listing persist filters)

## Steps to Test

1) Load up search form - http://localhost:3000/patterns/05-pages-location-listing/05-pages-location-listing.html
It should automatically change it to - http://localhost:3000/patterns/05-pages-location-listing/05-pages-location-listing.html?page=1
2) Perform a search:
Location - type in “salem”, select “Salem MA, USA” from autocomplete
Tags - check the box for Open Now
3) Click Submit
4) Verify results are correct, and results are showing correct data, such as the summary text
5) Check the box for Wheelchair accessible (so both wheelchair and Open are checked)
6) Click Submit
7) Click the “x” next to the “Open Now” filter in the listing header.
8) Click the “2” in the pager
9) Verify filters are still correct both in the form and in the filters
10) Click one of the result links (which should open mass.gov)
11) Click the Back button in the browser
The results should come back to page 2 of the results and it should have the filters applied for:
Location: Salem, MA, USA
Tags: Wheelchair Accessible
